File: 102/102_version1.java
class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        List<List<Integer>> res = new ArrayList<>();
        Queue<TreeNode> queue = new LinkedList<>();
        Queue<Integer> levelQueue = new LinkedList<>();
        queue.offer(root);
        levelQueue.offer(0);
        while(queue.size() > 0){
            TreeNode curr = queue.poll();
            int level = levelQueue.poll();
            if(curr!=null){
                if(res.size()==level) res.add(new ArrayList<>());
                res.get(level).add(curr.val);
                queue.offer(curr.left);levelQueue.offer(level+1);
                queue.offer(curr.right);levelQueue.offer(level+1);
            }
        }
        return res;
    }
}

File: 102/102_version1.py
class Solution:
    def levelOrder(self, root: TreeNode) -> List[List[int]]:
        queue = []
        res = []
        queue.append([0,root])
        while queue:
            level,node = queue.pop()
            if node:
                if level == len(res):
                    res.append([])
                res[level].append(node.val)
                queue.insert(0,[level+1,node.left])
                queue.insert(0,[level+1,node.right])
        return res
